src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -140,25 +144,74 @@ internal async ValueTask<(bool Authorized, RavenServer.AuthenticationStatus Stat
                 }
             }
 
-            if (CanAccessRoute(route, context, database?.Name, feature, out var authenticationStatus) == false)
+            if (CanAccessRoute(route, context, database?.Name, feature) == false)
             {
                 await UnlikelyFailAuthorizationAsync(context, database?.Name, feature, route.AuthorizationStatus);
-                return (false, authenticationStatus);
+                return (false, feature.Status);
             }
 
-            return (true, authenticationStatus);
+            if (feature.TwoFactorAuthRegistration is { HasLimits: true })
+            {
+                if (ValidateTwoFactorLimits(context, feature, out var msg) == false)
+                {
+                    if (LoggingSource.AuditLog.IsInfoEnabled)
+                    {
+                        var auditLog = LoggingSource.AuditLog.GetLogger("RequestRouter", "Audit");
+                        auditLog.Info($"Rejected request {context.Request.Method} {context.Request.GetFullUrl()} because: {msg}");
+                    }
+
+                    return (false, RavenServer.AuthenticationStatus.TwoFactorAuthFromInvalidLimit);
+                }
+            }
+
+            return (true, feature.Status);
+        }
+
+        private bool ValidateTwoFactorLimits(HttpContext context, RavenServer.AuthenticateConnection feature, out string msg)
+        {
+            if (context.Request.Cookies.TryGetValue(TwoFactorAuthentication.CookieName, out var cookieStr) == false)
+            {
+                msg = "Missing the csrf-cookie in the request";
```

Comment:
do we want to use TwoFactorAuthentication.CookieName here?

Reply:
Fixed